projectile.el

```diff
@@ -3551,25 +3551,28 @@ With a prefix ARG invokes `projectile-commander' instead of
 Invokes the command referenced by `projectile-switch-project-action' on switch.
 With a prefix ARG invokes `projectile-commander' instead of
 `projectile-switch-project-action.'"
-  (let ((switch-project-action (if arg
-                                   'projectile-commander
-                                 projectile-switch-project-action)))
-    (run-hooks 'projectile-before-switch-project-hook)
-    (let ((default-directory project-to-switch))
-      ;; use a temporary buffer to load PROJECT-TO-SWITCH's dir-locals before calling SWITCH-PROJECT-ACTION
-      (with-temp-buffer
-        (hack-dir-local-variables-non-file-buffer))
-      ;; Normally the project name is determined from the current
-      ;; buffer. However, when we're switching projects, we want to
-      ;; show the name of the project being switched to, rather than
-      ;; the current project, in the minibuffer. This is a simple hack
-      ;; to tell the `projectile-project-name' function to ignore the
-      ;; current buffer and the caching mechanism, and just return the
-      ;; value of the `projectile-project-name' variable.
-      (let ((projectile-project-name (funcall projectile-project-name-function
-                                              project-to-switch)))
-        (funcall switch-project-action)))
-    (run-hooks 'projectile-after-switch-project-hook)))
+  (if (projectile-project-p project-to-switch)
```

Comment:
I think the real thing to check for here would be whether the target dir exists at all.

Reply:
That's what I originally checked, however if the target dir still exists but is no longer a project (`.git` is gone, etc.) then there's still going to be issues.

Checking `projectile-project-p` should cover both cases.

Reply:
Let me know if you think there should be a check for both with more specific errors.  Or I can add more info to the single error like `Check the directory still exists and is a valid project` or something.

Reply:
I guess the current arrangement will be fine. Just one small suggestion - move the check to its own `when` that simply raises the `error` if needed. That way we won't have to nest all the code under the top `if`.